stats/statsvariables.cpp

```diff
@@ -0,0 +1,1674 @@
+		if (d->sac <= 0)
+			return invalid_value<int>();
+		return d->sac / 1000 / bin_size;
+	}
+};
+
+// "Imperial" SACs are annoying, since we have to bin to sub-integer precision.
+// We store cuft * 100 as an integer, to avoid troubles with floating point semantics.
+
+struct ImperialSACBinner : public IntBinner<ImperialSACBinner, IntBin> {
+	int bin_size;
+	ImperialSACBinner(double size)
+		: bin_size(lrint(size * 100.0))
+	{
+	}
+	QString name() const override {
+		QLocale loc;
+		return StatsTranslations::tr("in %1 %2/min steps").arg(loc.toString(bin_size / 100.0, 'f', 2),
+								       get_volume_unit());
+	}
+	QString format(const StatsBin &bin) const override {
+		int value = derived_bin(bin).value;
+		QLocale loc;
+		return StatsTranslations::tr("%1–%2").arg(loc.toString((value * bin_size) / 100.0, 'f', 2),
+							  loc.toString(((value + 1) * bin_size) / 100.0, 'f', 2));
+	}
+	QString unitSymbol() const override {
+		return get_volume_unit() + StatsTranslations::tr("/min");
+	}
+	QString formatLowerBound(const StatsBin &bin) const override {
+		int value = derived_bin(bin).value;
+		return QStringLiteral("%L1").arg((value * bin_size) / 100.0, 0, 'f', 2);
+	}
+	double lowerBoundToFloatBase(int value) const {
+		return static_cast<double>((value * bin_size) / 100.0);
+	}
+	int to_bin_value(const dive *d) const {
+		if (d->sac <= 0)
+			return invalid_value<int>();
+		return lrint(ml_to_cuft(d->sac) * 100.0) / bin_size;
+	}
+};
+
+MetricSACBinner metric_sac_binner2(2);
+MetricSACBinner metric_sac_binner5(5);
+MetricSACBinner metric_sac_binner10(10);
+ImperialSACBinner imperial_sac_binner1(0.1);
+ImperialSACBinner imperial_sac_binner2(0.2);
+ImperialSACBinner imperial_sac_binner4(0.4);
+ImperialSACBinner imperial_sac_binner8(0.8);
+
+struct SACType : public StatsVariableTemplate<StatsVariable::Type::Numeric> {
+	QString name() const override {
+		return StatsTranslations::tr("SAC");
+	}
+	QString unitSymbol() const override {
+		return get_volume_unit() + StatsTranslations::tr("/min");
+	}
+	int decimals() const override {
+		return prefs.units.volume == units::LITER ? 0 : 2;
+	}
+	std::vector<const StatsBinner *> binners() const override {
+		if (prefs.units.volume == units::LITER)
+			return { &metric_sac_binner2, &metric_sac_binner5, &metric_sac_binner10 };
+		else
+			return { &imperial_sac_binner1, &imperial_sac_binner2, &imperial_sac_binner4, &imperial_sac_binner8 };
+	}
+	double toFloat(const dive *d) const override {
+		if (d->sac <= 0)
+			return invalid_value<double>();
+		return prefs.units.volume == units::LITER ? d->sac / 1000.0 :
+							    ml_to_cuft(d->sac);
+	}
+	std::vector<StatsOperation> supportedOperations() const override {
+		return { StatsOperation::Median, StatsOperation::Mean, StatsOperation::TimeWeightedMean };
+	}
+};
+
+// ============ Water and air temperature, binned in 2, 5, 10, 20 °C/°F bins ============
+
+// We use the same binner for °C and °F. We simply output the current temperature unit.
```

Comment:
that's... surprising... but I guess it qualifies as "good enough"

Reply:
The unit is now made explicit.